src/views/Alteration.vue

```diff
@@ -37,30 +39,58 @@ import { SessionStorageKeys } from 'sbc-common-components/src/util/constants'
   }
 })
 export default class Alteration extends Mixins(LegalApiMixin, FilingTemplateMixin) {
-  // Getter definition for static type checking.
+  // Global getters
   @Getter isRoleStaff!: boolean
+  @Getter getOrgPeople!: OrgPersonIF[]
+  @Getter getShareClasses!: ShareClassIF[]
 
-  @State(state => state.stateModel.addPeopleAndRoleStep.orgPeople)
-  readonly orgPersonList: OrgPersonIF[]
+  /** The id of the IA filing to alter. */
+  // private get filingId (): number {
+  //   return +this.$route.query['filing-id']
+  // }
 
-  @State(state => state.stateModel.createShareStructureStep.shareClasses)
-  readonly shareClasses: ShareClassIF[]
+  /** True if user is authenticated. */
+  private get isAuthenticated (): boolean {
+    return Boolean(sessionStorage.getItem(SessionStorageKeys.KeyCloakToken))
+  }
 
-  private mounted (): void {
-    console.log('*** Alteration view is mounted') // eslint-disable-line no-console
+  /** Called when this component is mounted. */
+  private async mounted (): Promise<void> {
+    // do not proceed if FF is disabled
+    if (!featureFlags.getFlag('alteration-ui-enabled')) {
+      alert('Alterations are under contruction. Please check again later.')
+      return
+    }
+
+    // do not proceed if we are not anthenticated
+    // (this component will be re-mounted after authentication)
     if (!this.isAuthenticated) return
 
-    // If a user (not staff) tries this url directly, return them to the Manage Businesses dashboard.
+    // do not proceed if user is not staff
     const isStaffOnly = this.$route.matched.some(r => r.meta?.isStaffOnly)
     if (isStaffOnly && !this.isRoleStaff) {
+      alert('Only staff can alter a business.')
+      // redirect to the Manage Businesses dashboard
       const manageBusinessUrl = `${sessionStorage.getItem('AUTH_URL')}business`
       window.location.assign(manageBusinessUrl)
+      return
```

Comment:
As triggered by your comment, I added a return here also. This is to handle slow redirections (otherwise the component keeps loading).

Reply:
ah good call!